components/provisioning/org.wso2.carbon.identity.provisioning/src/test/java/org/wso2/carbon/identity/provisioning/ProvisioningUtilTest.java

```diff
@@ -0,0 +1,385 @@
+        inboundClaimMap.put(remoteClaimUri, "testuser3");
+        inboundClaimMap.put(remoteClaimUri + "1", "testuser4");
+
+        inboundClaimMap1.put(localClaimUri, "testuser");
+        inboundClaimMap1.put(localClaimUri + "1", "testuser1");
+
+        outboundClaimValueMappings = new HashMap();
+        outboundClaimValueMappings.put(mappings[0], values);
+        outboundClaimValueMappings.put(mappings[1], values1);
+    }
+
+    @Test(dataProvider = "getClaimValueData")
+    public void testGetClaimValues(Map<ClaimMapping, List<String>> attributeMap,
+                                   String claimUri, String userStoreDomainName) throws Exception {
+
+        assertNotNull(ProvisioningUtil.getClaimValues(attributeMap, claimUri, userStoreDomainName));
+
+    }
+
+
+    @Test
+    public void testSetClaimValue() throws Exception {
+
+        ProvisioningUtil.setClaimValue(localClaimUri + "2", attributeMap, values1);
+        List<String> resultValues = attributeMap.get(ClaimMapping.build(localClaimUri + "2", remoteClaimUri, null,
+                false));
+        assertEquals(resultValues, values1);
+
+        ProvisioningUtil.setClaimValue(localClaimUri + "2", attributeMap1, values1);
+        List<String> resultValues1 = attributeMap1.get(ClaimMapping.build(localClaimUri + "2", remoteClaimUri, null,
+                false));
+        assertNull(resultValues1);
+
+    }
+
+
+    @Test
+    public void testGetAttributeValue() {
+
+        ProvisioningEntity provisioningEntity = new ProvisioningEntity(USER, "testEntityName", POST, attributeMap);
+        String value = ProvisioningUtil.getAttributeValue(provisioningEntity, localClaimUri + "1");
+        assertEquals(value, values1.get(0));
+
+        ProvisioningEntity provisioningEntity1 = new ProvisioningEntity(USER, "testEntityName", POST, attributeMap1);
+        String value1 = ProvisioningUtil.getAttributeValue(provisioningEntity1, localClaimUri + "1");
+        assertNull(value1);
+
+        ProvisioningEntity provisioningEntity2 = new ProvisioningEntity(USER, "testEntityName", POST, attributeMap1);
+        String value2 = ProvisioningUtil.getAttributeValue(provisioningEntity2, localClaimUri + "1");
+        assertNull(value2);
+    }
+
+
+    @Test(dataProvider = "getInboundClaimData")
+    public void testGetMappedClaims(String outboundClaimDialect,
+                                    Object inboundClaimValueMap, Object inboundClaimMappings,
+                                    Object outboundClaimValueMappings, String tenantDomain) throws Exception {
+
+        Map<ClaimMapping, List<String>> result = null;
+        result = ProvisioningUtil.getMappedClaims(outboundClaimDialect,
+                (Map<String, String>) inboundClaimValueMap, (ClaimMapping[]) inboundClaimMappings,
+                (Map<ClaimMapping, List<String>>)
+                        outboundClaimValueMappings, tenantDomain);
+        assertNotNull(result);
```

Comment:
Other than asserting on not null, better to assert on the content as well. This will not be a good test just because you assert the content is not null. Better to check whether the correct content is there inside. This applies to all above tests :)